### PR TITLE
[fundamental]Github Token should use along with permissions.

### PR DESCRIPTION
**Pull request by crazygao**
# Description

Remove currently unused GITHUB_TOKEN.
For places that need this token, add necessary permissions at job level.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: .github/workflows/promptflow-executor-e2e-test.yml [hunk 1]
```diff
@@ -161,5 +161,4 @@ jobs:
         osVersion: ubuntu-latest
         pythonVersion: 3.9
         coverageThreshold: 70
-        token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (zhengfeiwang) [inline .github/workflows/promptflow-executor-e2e-test.yml:164]: as the CI passed, I'm curious on why this workflow can pass without specify `token`?

File: .github/workflows/flowdag_schema_check.yml [hunk 2]
```diff
@@ -22,7 +22,6 @@ jobs:
         shell: pwsh
         env:
           PYTHONPATH: ${{ github.workspace }}/src/promptflow
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (zhengfeiwang) [inline .github/workflows/flowdag_schema_check.yml:25]: so we don't use this inside our script?

File: .github/actions/step_publish_test_results/action.yml [hunk 3]
```diff
@@ -19,9 +19,6 @@ inputs:
     required: false
```
Comment (zhengfeiwang) [inline .github/actions/step_publish_test_results/action.yml:19]: `create_promptflow_release_tag` also have some references on `GITHUB_TOKEN`, may take a look

**Comment by zhengfeiwang** [comment]
better to summarize our usage of `GITHUB_TOKEN` in PR description, so that we can better report the changes.

File: .github/actions/step_publish_test_results/action.yml (hunk 3 above)
Comment (crazygao) [reply .github/actions/step_publish_test_results/action.yml:19]: permission is already provided in that file.

File: .github/workflows/flowdag_schema_check.yml (hunk 2 above)
Comment (crazygao) [reply .github/workflows/flowdag_schema_check.yml:25]: this is used in previous version of publish. it is not used now.

File: .github/workflows/promptflow-executor-e2e-test.yml (hunk 1 above)
Comment (crazygao) [reply .github/workflows/promptflow-executor-e2e-test.yml:164]: we don't use the token at all.

**Review by zhengfeiwang — APPROVED** [review]

**Review by 16oeahr — DISMISSED** [review]

File: .github/workflows/tools_release_tag.yml [hunk 4]
```diff
@@ -21,6 +21,8 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
```
Comment (16oeahr) [inline .github/workflows/tools_release_tag.yml:25]: do we need to give "actions: read" permission?

Comment (zhengfeiwang) [reply .github/workflows/tools_release_tag.yml:25]: I understand `write` permission should imply `read`, so we don't need to specify `read, write` here.

Comment (zhengfeiwang) [reply .github/workflows/tools_release_tag.yml:25]: unresolve this to make it a blocker, we still need to confirm something with this GH change

**Review by wangchao1230 — APPROVED** [review]

File: .github/workflows/tools_release_tag.yml (hunk 4 above)
Comment (crazygao) [reply .github/workflows/tools_release_tag.yml:25]: resolve, we have test that current permission is sufficient.

**Review by wangchao1230 — APPROVED** [review]

**Review by zhengfeiwang — APPROVED** [review]

**Pull request merged**